### DISCUSSION TITLE
how to backup and restore the applications in case delete it by mistake

### DISCUSSION THREAD

**Discussion by backearth1** [langgenius/dify#4380] [Help] [answered]
self hosted deployment, many thanks

**Comment by patryk20120** [comment, marked as answer]
You can export / import DSL file in the apps studio context menu

**Comment by sergio-campos-trebolit** [comment]
Is there a way to do this via API? I was thinking about making my CI/CD pipeline in a non-interactive mode. Do you know if it is possible? Any suggestions or documentation would be greatly appreciated.

**Reply by patryk20120** [reply]
It is not possible by API.

**Reply by sergio-campos-trebolit** [reply]
But there is a console API. Maybe it's not documented.
[app.py](https://github.com/langgenius/dify/blob/main/api/controllers/console/app/app.py#L269)
Additionally, **two environment variable**s are mentioned in: [login.py](https://github.com/langgenius/dify/blob/main/api/libs/login.py#L55)
- `ADMIN_API_KEY`
- `ADMIN_API_KEY_ENABLE`

**Reply by sergio-campos-trebolit** [reply]
I made a pull request with a little tutorial :) https://github.com/langgenius/dify/pull/5006

**Reply by patryk20120** [reply]
So I have changed my mind. It is possible! :)
